src/tls_openssl.c

```diff
@@ -218,6 +230,94 @@ int tls_error(tls_t *tls)
     return tls->lasterror;
 }
 
+static GENERAL_NAMES *_tls_cert_get_names(xmpp_conn_t *conn)
+{
+    GENERAL_NAMES *names = NULL;
+    X509 *cert = _tls_cert_read(conn);
+    if (!cert)
+        return NULL;
+    int san = X509_get_ext_by_NID(cert, NID_subject_alt_name, 0);
+    X509_EXTENSION *san_ext = X509_get_ext(cert, san);
+    if (!san_ext)
+        goto OUT;
+    ASN1_OCTET_STRING *data = X509_EXTENSION_get_data(san_ext);
+    if (!data)
+        goto OUT;
+    const unsigned char *d = ASN1_STRING_get0_data(data);
+    if (!d)
+        goto OUT;
+    names = d2i_GENERAL_NAMES(NULL, &d, ASN1_STRING_length(data));
+OUT:
+    X509_free(cert);
+    return names;
+}
+
+static int _tls_name_to_xmppaddr(GENERAL_NAME *name, char **res)
```

Comment:
Please add a comment about return value. It is clear from reading the code, but not obvious from the prototype. The below code uses it as a boolean, but 0 is actually a successful value.